include/utility.hpp

```diff
@@ -433,7 +433,11 @@ template <class SH_CORE> class TShardsVar {
   void destroy() {
     for (auto it = _shardsArray.rbegin(); it != _shardsArray.rend(); ++it) {
       auto blk = *it;
-      blk->cleanup(blk);
+      auto errors = blk->cleanup(blk);
+      if (errors.code != 0) {
```

Comment:
I think this `0` value should be named for clarity. Maybe something like `NO_ERROR`.

Alternatively it could be an inlined function like `errors.hasError()`.

Reply:
Sorted